var/spack/repos/builtin/packages/perl/package.py

```diff
@@ -372,8 +368,7 @@ def setup_build_environment(self, env):
 
         spec = self.spec
 
-        if (spec.version <= Version('5.34.0')
-            and spec.platform == 'darwin'
+        if (spec.satisfies('@:5.34 platform=darwin')
```

Comment:
I think you correctly "ported" the code as it was before. Anyway, apparently even on newer versions of perl this change is required.

I would take the chance to improve this condition in this PR. I will try to investigate a bit to understand what are the perl versions requiring this "patch".

Reply:
The `@:5.34` includes everything through `5.34.99999` (i.e. the failing perl implementation) so it's a superset of your patch. It *probably* won't hurt to make this change to `MACOSX_DEPLOYMENT_TARGET` for newer versions of Perl, but I've been hoping the developers would fix it sooner.

Reply:
Ah you're right, I always confuse

```python
>>> Version("3.4.1") <= Version("3.4")
False
>>> Version("3.4.1").satisfies(Version("3.4"))
True
```

Thanks!